GrpcCodeGen fails with Quarkus 3.31 on Java 22 + 23

### Describe the bug

In the build process while generating Grpc server code using Java 22 or Java 23, I obtain

> [ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.31.1:generate-code (default) on project <X>: 
> 
>  Process exited abnormally (pid 478538) with exit code 1 with error output:
> [ERROR]  > NOTE: Picked up JDK_JAVA_OPTIONS: --sun-misc-unsafe-memory-access=allow -Dsun.stdout.encoding=UTF-8
> [ERROR]  > Unrecognized option: --sun-misc-unsafe-memory-access=allow
> [ERROR]  > Error: Could not create the Java Virtual Machine.
> [ERROR]  > Error: A fatal exception has occurred. Program will exit.
> [ERROR]  > --q-grpc_out: protoc-gen-q-grpc: Plugin failed with status code 1.
> [ERROR] -> [Help 1]

This seems to be due to [these lines](https://github.com/quarkusio/quarkus/commit/8ea7bbbc034b75324cc225b3b7b15f6ad75a2b68#diff-d1d35e04762cd6609f8d401f5b9b18e8ded271558422e6f67327681ecae25c7e) in GrpcCodeGen:

```
        //Do NOT set this property on Java 17, as it will fail with "unrecognized option":
        if (Runtime.version().feature() > 21 && !existingValue.contains("--sun-misc-unsafe-memory-access=")) {
            sb.append("--sun-misc-unsafe-memory-access=allow ");
        }
```
The JVM Property is however unrecognized in Java 22 and Java 23 as well and should therefore be appended only for Java Version 24 onwards, similar to the implementation in [JvmChecksRecorder](https://github.com/quarkusio/quarkus/commit/22d9fc84e66ab33912bfdb6c4acc6240e38ca972#diff-eaafa3e2b6f3bfd8139d421a44b64be349e21180b436dde50359fa086d4e0cce)

```
    public static boolean isUnsafeMemoryAccessAllowed() {
        if (Runtime.version().feature() < 24) {
            //Versions before Java 24 would not complain about the use of Unsafe.
            //Also, setting `--sun-misc-unsafe-memory-access=allow` isn't possible (not a valid argument) before Java 24.
            return true;
        }
        RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
        List<String> arguments = runtimeMxBean.getInputArguments();
        return arguments.contains("--sun-misc-unsafe-memory-access=allow");
    }
```

### Expected behavior

GrpcCodeGen should work with java 22 and 23  

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

22

### Quarkus version or git rev

3.31.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/grpc/codegen/src/main/java/io/quarkus/grpc/codegen/GrpcCodeGen.java
+++ b/extensions/grpc/codegen/src/main/java/io/quarkus/grpc/codegen/GrpcCodeGen.java
@@ -235,8 +235,8 @@ public class GrpcCodeGen implements CodeGenProvider {
             sb.append("-Dsun.stdout.encoding=UTF-8 ");
         }
 
-        //Do NOT set this property on Java 17, as it will fail with "unrecognized option":
-        if (Runtime.version().feature() > 21 && !existingValue.contains("--sun-misc-unsafe-memory-access=")) {
+        //Do NOT set this property on Java versions before 24, as it will fail with "unrecognized option":
+        if (Runtime.version().feature() >= 24 && !existingValue.contains("--sun-misc-unsafe-memory-access=")) {
             sb.append("--sun-misc-unsafe-memory-access=allow ");
         }
         sb.append(existingValue);

